modules/class-simplechart-save.php

```diff
@@ -91,30 +98,14 @@ function do_save_post( $post ) {
 		}
 
 		// handle base64 image string if provided
-		if ( ! empty( $_POST['simplechart-png-string'] ) ) {
-			$this->_save_chart_image( $post, $_POST['simplechart-png-string'], $this->_default_img_type );
+		if ( ! empty( $_POST['save-previewImg'] ) ) {
+			$this->_save_chart_image( $post, $_POST['save-previewImg'], $this->_default_img_type );
 		}
 
-		// sanitize and validate JSON formatting of chart data
-		$json_data = $this->_validate_json( stripslashes( $_POST['simplechart-data'] ) );
-		if ( $json_data ) {
-			update_post_meta( $post->ID, 'simplechart-data',  $json_data );
-		}
-
-		// validate template HTML fragment
-		$template_fragment = $this->validate_template_fragment( stripslashes( $_POST['simplechart-template'] ) );
-		if ( $template_fragment ) {
-			update_post_meta( $post->ID, 'simplechart-template',  $template_fragment );
-		}
-
-		// save chart URL if provided
-		if ( ! empty( $_POST['simplechart-chart-url'] ) ) {
-			update_post_meta( $post->ID, 'simplechart-chart-url',  esc_url( $_POST['simplechart-chart-url'] ) );
-		}
-
-		// save chart ID if provided
-		if ( ! empty( $_POST['simplechart-chart-id'] ) ) {
-			update_post_meta( $post->ID, 'simplechart-chart-id',  sanitize_text_field( $_POST['simplechart-chart-id'] ) );
+		foreach ( $this->meta_field_names as $field ) {
+			if ( ! empty( $_POST[ 'save-' . $field ] ) ) {
+				update_post_meta( $post->ID, 'save-' . $field, $_POST[ 'save-' . $field ] );
```

Comment:
Maybe? It passes to `update_metadata()` which passes to `sanitize_meta()` which applies filters based on the meta type and meta key. It further runs through some filters for the meta type. You can always see what VIP has to say. I suspect they'll ask you to sanitize early.